tools/esp-api/espservice.cpp

```diff
@@ -25,45 +25,51 @@ using namespace std;
 
 int EspService::sendRequest()
 {
-
     int responseCode = -1;
     StringBuffer req,res,status;
-    req.append(formArgs);
+    req.append(reqString);
 
     Owned<IHttpClientContext> httpctx = getHttpClientContext();
     Owned <IHttpClient> httpclient = httpctx->createHttpClient(NULL, url);
 
-    if(!streq(username, "") && !streq(password, ""))
+    if(!isEmptyString(username))
     {
         httpclient->setUserID(username);
+    }
+    if(!isEmptyString(password))
+    {
         httpclient->setPassword(password);
     }
 
     if(streq(reqType, "json"))
     {
-       responseCode = httpclient->sendRequest("POST","application/json",req,res, status ,true);
+       responseCode = httpclient->sendRequest("POST", "application/json", req, res, status, true);
     }
     else if(streq(reqType, "form"))
     {
-       responseCode = httpclient->sendRequest("POST","application/x-www-form-urlencoded",req,res, status ,true);
+       responseCode = httpclient->sendRequest("POST", "application/x-www-form-urlencoded", req, res, status, true);
     }
     else if(streq(reqType, "xml"))
     {
-       responseCode = httpclient->sendRequest("POST","text/xml",req,res, status ,true);
+       responseCode = httpclient->sendRequest("POST", "text/xml", req, res, status, true);
     }
 
     if(responseCode != 0 )
     {
-        cerr << "Enter a valid esp target" << endl;
+        cerr << "Response code:" << responseCode << " , Enter a valid esp target" << endl;
```

Comment:
To be consistent with the error message on 65, consider `cerr << "Response code " << responseCode << ": Enter a valid esp target" << endl`